src/is_master.rs

```diff
@@ -34,72 +43,166 @@ pub(crate) fn is_master_command(api: Option<&ServerApi>) -> Command {
 pub(crate) async fn run_is_master(
     command: Command,
     conn: &mut Connection,
-) -> Result<IsMasterReply> {
+    handler: &Option<Arc<dyn SdamEventHandler>>,
+) -> Result<HelloReply> {
     if !command.name.eq_ignore_ascii_case("ismaster") && !command.name.eq_ignore_ascii_case("hello")
     {
         return Err(ErrorKind::Internal {
             message: format!("invalid ismaster command: {}", command.name),
         }
         .into());
     }
+
+    if let Some(ref handler) = handler {
```

Comment:
I did a brief scan and I couldn't find any places where it's constructed as `None`, so I think we can just make `round_trip_time` non-optional if we want. If that's the only blocker here then I think it's probably best for us to make that change and move this logic into the monitor.

Reply:
Ah I see. So I think that line doesn't actually ever execute, since it only executes if the reply is `Some(Ok(_))` and the server type is unknown, but the server type is unknown only if the `IsMasterReply` is `None` or `Err`. Also, if the reply is `None` or `Err`, then the round trip time of the `ServerDescription` will be `None` as well, so I think we do abide by the spec anyways and we're still safe to remove the `Option` from `IsMasterReply`.

Only reason I push this is it helps make progress towards RUST-726 and also due to the potential races involved with having the logic here I mentioned in another comment.

Reply:
Cool, made the change and it doesn't seem to impact anything. I did need to adjust the return type of `run_is_master` to return the `round_trip_time` in the error case.